docs/language_guide.md

```diff
@@ -659,20 +705,27 @@ Wallet :: (owner) {
 
 Flint introduces the concept of **caller capabilities**. While traditional computer programs have an entry point (the _main_ function), smart contracts do not. After a contract is deployed on the blockchain, its code does not run until an Ethereum transaction is received. Smart contracts are in fact more akin to RESTful web services presenting API endpoints. It is important to prevent unauthorized parties from calling sensitive functions.
 
-In Flint, functions of a contract are declared within caller capability blocks, which restrict which users/contracts are allowed to call the enclosed functions.
-```swift
-// Only the manager of the Bank can call "clear".
-Bank :: (manager) { // manager is a state property.
-  func clear(address: Address) {
-    // body
-  }
-}
-```
-Caller capabilities can be any property declared in the contract's declaration, as long as their type is `Address` or `[Address]`.
+Solidity uses function modifiers to insert dynamic checks in functions, which can for instance abort unauthorised calls. However, it is easy to forget to specify these checks, as the language does not require programmers to write them.
+Having a language construct which protects functions from invalid calls could require programmers to systematically think about which parties should be able to call the functions they are about to define.
 
-Note: The special caller capability any allows `any` caller to execute the function in the group.
+In Flint, functions of a contract are declared within caller capability blocks, which protect the functions from invalid access.
 
-Calls to Flint functions are validated both at compile-time and runtime.
+### Caller Group
+Caller Groups consist of a list of Caller Members enclosed in parentheses. These Caller Members can be a function of type `Address -> Bool` or `() -> Address`, a contract state property of the types `Address`, `[Address]`, `[Address: T]` or `Address[n]`, or the special `any` keyword that denotes all addresses. For example, `(any)`, `(admin)`, `(owners, manager)` are all valid Caller Groups where `admin` and `manager` refer to a state property of type `Address` and `owners` refers to a state property of type `[Address]`
```

Comment:
`[T: Address]` rather than `[Address: T]` is the dictionary type supported.

Reply:
You can tell iI didn't write that one!